File: app/Models/Tenant/Vehicle.php
<?php

namespace App\Models\Tenant;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\SoftDeletes;

class Vehicle extends Model
{
    use SoftDeletes;

    protected $connection = 'society';

    protected $fillable = [
        'user_id',
        'vehicle_type',
        'registration_number',
        'model',
        'color',
        'year',
        'parking_slot',
        'status',
    ];

    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class);
    }

    public function scopeActive($query)
    {
        return $query->where('status', 'active');
    }
}

File: mobile/lib/features/profile/data/vehicle.dart
class Vehicle {
  Vehicle({
    required this.id,
    required this.vehicleType,
    required this.registrationNumber,
    this.model,
    this.color,
    this.year,
    this.parkingSlot,
    required this.status,
  });

  factory Vehicle.fromJson(Map<String, dynamic> json) {
    return Vehicle(
      id: json['id'] as int,
      vehicleType: json['vehicle_type'] as String,
      registrationNumber: json['registration_number'] as String,
      model: json['model'] as String?,
      color: json['color'] as String?,
      year: json['year'] as int?,
      parkingSlot: json['parking_slot'] as String?,
      status: json['status'] as String? ?? 'active',
    );
  }

  final int id;
  final String vehicleType;
  final String registrationNumber;
  final String? model;
  final String? color;
  final int? year;
  final String? parkingSlot;
  final String status;
}
